pkg/ebpfsyncer/ebpfsyncer.go

```diff
@@ -80,93 +84,176 @@ func (e *ebpfSingleton) SyncInterfaceIngressRules(
 		}()
 	}
 
+	// Create a new manager if none exists.
+	if err := e.createNewManager(); err != nil {
+		return err
+	}
+
 	// For delete operations, detach all interfaces and run a cleanup, set managed interfaces and the
 	// manager to empty / nil values, then return.
-	// TODO: This should purge all XDP attachments on all interfaces even if we are not managing them.
-	// Alternatively, we need something like method XDPInterfacePurge() that would purse all unmanaged interfaces
-	// both from the rules and from the interface attachements.
 	if isDelete {
-		logger.Info("Running detach and eBPF cleanup operation")
-		for intf := range e.managedInterfaces {
-			_, err := e.c.IngressNodeFwAttach([]string{intf}, true)
-			if err != nil {
-				return err
-			}
-		}
-		e.managedInterfaces = make(map[string]struct{})
-		err := e.c.CleaneBPFObjs()
-		if err != nil {
-			return err
-		}
-		e.c = nil
+		return e.resetAll()
+	}
 
-		return nil
+	// Detach unmanaged interfaces that were previously managed.
+	if err := e.detachUnmanagedInterfaces(ifaceIngressRules); err != nil {
+		return err
 	}
 
-	// Create a new manager if none exists.
+	// Attach interfaces which shall now be managed.
+	if err := e.attachNewInterfaces(ifaceIngressRules); err != nil {
+		return err
+	}
+
+	// Attach failsafe rules to the entire ruleset.
+	if err := e.attachFailsafeRules(ifaceIngressRules); err != nil {
+		return err
+	}
+
+	// Load IngressNodeFirewall Rules (this is idempotent and will add new rules and purge rules that shouldn't exist).
+	if err := e.loadIngressNodeFirewallRules(ifaceIngressRules); err != nil {
+		return err
+	}
+	return nil
+}
+
+// getBPFMapContent lists the content of the current BPF map. Used for unit testing only.
+func (e *ebpfSingleton) getBPFMapContent() (map[nodefwloader.BpfLpmIpKeySt]nodefwloader.BpfRulesValSt, error) {
```

Comment:
same here pls add Test to the name to make it clear its ut only

Reply:
done